pkg/certifier/osv/osv_test.go

```diff
@@ -0,0 +1,336 @@
+	text4ShelPackage.Purl = "pkg:maven/org.apache.commons/commons-text@1.9"
+
+	text4shell := &certifier.Component{
+		CurPackage:  text4ShelPackage,
+		DepPackages: []*certifier.Component{},
+	}
+
+	log4j := &certifier.Component{
+		CurPackage:  log4JPackage,
+		DepPackages: []*certifier.Component{},
+	}
+
+	secondLevel := &certifier.Component{
+		CurPackage:  secondLevelPackage,
+		DepPackages: []*certifier.Component{text4shell},
+	}
+
+	rootComponent := &certifier.Component{
+		CurPackage:  rootPackage,
+		DepPackages: []*certifier.Component{secondLevel, log4j},
+	}
+	tests := []struct {
+		name          string
+		rootComponent *certifier.Component
+		want          []*processor.Document
+		wantErr       bool
+	}{{
+		name:          "query and generate attestation for OSV",
+		rootComponent: rootComponent,
+		want: []*processor.Document{
+			{
+				Blob:   []byte(text4ShellVulAttestation),
+				Type:   processor.DocumentITE6Vul,
+				Format: processor.FormatJSON,
+				SourceInformation: processor.SourceInformation{
+					Collector: INVOC_URI,
+					Source:    INVOC_URI,
+				},
+			},
+			{
+				Blob:   []byte(SecondLevelVulAttestation),
+				Type:   processor.DocumentITE6Vul,
+				Format: processor.FormatJSON,
+				SourceInformation: processor.SourceInformation{
+					Collector: INVOC_URI,
+					Source:    INVOC_URI,
+				},
+			},
+			{
+				Blob:   []byte(log4JVulAttestation),
+				Type:   processor.DocumentITE6Vul,
+				Format: processor.FormatJSON,
+				SourceInformation: processor.SourceInformation{
+					Collector: INVOC_URI,
+					Source:    INVOC_URI,
+				},
+			},
+			{
+				Blob:   []byte(rootVulAttestation),
+				Type:   processor.DocumentITE6Vul,
+				Format: processor.FormatJSON,
+				SourceInformation: processor.SourceInformation{
+					Collector: INVOC_URI,
+					Source:    INVOC_URI,
+				},
+			},
+		},
+		wantErr: false,
+	}}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			o := NewOSVCertificationParser()
+			collectedDocs := []*processor.Document{}
+			docChan := make(chan *processor.Document, 1)
+			errChan := make(chan error, 1)
+			defer close(docChan)
+			defer close(errChan)
+			go func() {
+				errChan <- o.CertifyVulns(ctx, tt.rootComponent, docChan)
+			}()
+			numCollectors := 1
+			certifiersDone := 0
+			for certifiersDone < numCollectors {
+				select {
+				case d := <-docChan:
+					collectedDocs = append(collectedDocs, d)
+				case err := <-errChan:
+					if (err != nil) != tt.wantErr {
+						t.Errorf("g.RetrieveArtifacts() error = %v, wantErr %v", err, tt.wantErr)
+						return
+					}
+					certifiersDone += 1
+				}
+			}
+			// Drain anything left in document channel
+			for len(docChan) > 0 {
+				d := <-docChan
+				collectedDocs = append(collectedDocs, d)
+			}
+			for i := range collectedDocs {
+				if !docEqual(t, collectedDocs[i], tt.want[i]) {
+					t.Errorf("g.RetrieveArtifacts() = %v, want %v", string(collectedDocs[i].Blob), string(tt.want[i].Blob))
```

Comment:
hmm would the datetime not match? might me missing somethign

Reply:
Depending on when the unit test is run, the datatime would be different. So this just changes them to a hardcoded value so that the comparison passes.